### PR TITLE
Add codemirror to Python Lab

### PR DESCRIPTION
**Pull request by molly-moen**
This PR adds [codemirror 6](https://codemirror.net/) with python language support to Python Lab. Right now the code is stored in redux and mirrored to a div for testing:

Screenshot 2023-12-14 at 9 32 36 AM

The main work here was getting codemirror running in a typescript component. I needed to update our TypeScript module and module resolution values to `node16`. This seems like the [recommended version](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext) given we use Node 18.

For now I'm directly using some of Java Lab's codemirror configuration, which we can probably refactor out into lab2 if we decide to keep that configuration for both labs.

## Links

- jira ticket: [CT-188](https://codedotorg.atlassian.net/browse/CT-188)

## Testing story
Tested the new lab locally. I also spot checked Music Lab to ensure changing the TypeScript module and module resolution configuration did not mess anything up.

## Follow-up work
I will follow this up with a PR that runs python in the browser.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/tsconfig.json [hunk 1]
```diff
@@ -11,9 +11,9 @@
     "jsx": "react",
     "target": "es5",
     "lib": ["ES2022", "DOM"],
-    "module": "es6",
+    "module": "node16",
```
Comment (molly-moen) [inline apps/tsconfig.json:14]: @sanchitmalhotra126 I tagged you on this PR because I changed some TypeScript configuration values.

Comment (sanchitmalhotra126) [reply apps/tsconfig.json:14]: LGTM 👍 out of curiosity, looking at the tsconfig [docs](https://www.typescriptlang.org/tsconfig#moduleResolution) I see 
> These moduleResolution values, when combined with the corresponding [module](https://www.typescriptlang.org/tsconfig#module) values, picks the right algorithm for each resolution based on whether Node.js will see an import or require in the output JavaScript code.

Does this allow TS to resolve modules with `require` in our code now?

**Review by sanchitmalhotra126 — APPROVED** [review]
tsconfig updates look good to me! (and cool to see Python Lab coming together!)

File: apps/tsconfig.json (hunk 1 above)
Comment (molly-moen) [reply apps/tsconfig.json:14]: I think maybe yes? But I'm not sure

File: apps/src/pythonlab/PythonlabView.tsx [hunk 2]
```diff
@@ -1,8 +1,16 @@
 // Pythonlab view
 import React from 'react';
+import PythonEditor from './PythonEditor';
+import moduleStyles from './pythonlab-view.module.scss';
+import SidePanel from './SidePanel';
 
 const PythonlabView: React.FunctionComponent = () => {
-  return <div>Welcome to Python lab!</div>;
+  return (
+    <div className={moduleStyles.pythonLab}>
```
Comment (ebeastlake) [inline apps/src/pythonlab/PythonlabView.tsx]: Super nit: If we standardize on `Pythonlab` and `pythonlab`, should we use that same convention here?

**Review by ebeastlake — APPROVED** [review]

File: apps/src/pythonlab/PythonlabView.tsx (hunk 2 above)
Comment (molly-moen) [reply apps/src/pythonlab/PythonlabView.tsx]: yep we should, will update!

**Pull request merged**